infrastructures/kubernetes/src/main/java/org/eclipse/che/workspace/infrastructure/kubernetes/namespace/KubernetesDeployments.java

```diff
@@ -388,19 +392,46 @@ public void onClose(KubernetesClientException cause) {
   }
 
   private void handleStartingPodStatus(CompletableFuture<Void> podRunningFuture, Pod pod) {
-    if (POD_STATUS_PHASE_RUNNING.equals(pod.getStatus().getPhase())) {
-      podRunningFuture.complete(null);
+    PodStatus status = pod.getStatus();
+    String podPhase = status.getPhase();
+    if (POD_STATUS_PHASE_RUNNING.equals(podPhase)) {
```

Comment:
I just wonder if these changes affects better error messages in case of plugin broker container error.
Seems they do not, please consider adding an invocation of `waitRunning` method in `DeployBroker.java` if we can do such easily.

Reply:
P.S. Such a case with a bad error message in case of plugin broker error is misconfigured Che in self-signed TLS https://mattermost.eclipse.org/eclipse/pl/aw5yt8zdd78zbpgfk3xx3y6j4h
```
Pulling image "eclipse/che-init-plugin-broker:v0.24"
Successfully pulled image "eclipse/che-init-plugin-broker:v0.24"
Created container che-init-plugin-broker-v0-24
Started container che-init-plugin-broker-v0-24
Error: Failed to run the workspace: "Plugins installation process timed out"
```
<- in fact, Plugin Broker exited and failed to connect to Che Server via WebSocket

Reply:
While this is a very good point, let's not broaden the scope of this issue. Would you please raise another issue for this?

Reply:
> Would you please raise another issue for this?

It totally makes sense! Will do

Reply:
It's here https://github.com/eclipse/che/issues/15456